src/pool.js

```diff
@@ -46,11 +47,18 @@ export default class Pool extends EventEmitter {
     }
 
     const job = this.jobQueue.shift();
+
+    if (job.isAborted) {
```

Comment:
I would add a `job.once('abort', () => { /* unqueue job */ })` to `queueJob()` instead.

Reply:
Might need to add a new method: `dropJob(job)`